pkg/pdutil/pd.go

```diff
@@ -492,43 +503,51 @@ func (p *PdController) RemoveSchedulers(ctx context.Context) (undo utils.UndoFun
 	if err != nil {
 		return
 	}
-
-	undo = p.makeUndoFunctionByConfig(clusterConfig{scheduler: removedSchedulers, scheduleCfg: scheduleCfg})
-	log.Debug("saved PD config", zap.Any("config", scheduleCfg))
-
-	disableMergeCfg := make(map[string]interface{})
-	for _, cfgKey := range pdRegionMergeCfg {
-		value := scheduleCfg[cfgKey]
-		if value == nil {
+	disablePDCfg := make(map[string]interface{})
+	for cfgKey, cfgVal := range expectPDCfg {
+		value, ok := scheduleCfg[cfgKey]
+		if !ok {
 			// Ignore non-exist config.
 			continue
 		}
-		// Disable region merge by setting config to 0.
-		disableMergeCfg[cfgKey] = 0
+		switch v := cfgVal.(type) {
+		case bool:
+			disablePDCfg[cfgKey] = "false"
+		case int:
+			limit := int(value.(float64))
+			disablePDCfg[cfgKey] = int(math.Min(40, float64(limit*len(stores)))) * v
+		}
```

Comment:
ok i finally got this logic, and i gotta say this is very convoluted to implement this behavior. i suggest:

1. introduce an enum to specify the expected behavior, instead of using the magic values `0`, `1` and `false`.

    ```go
    type pauseConfigExpectation uint8
    const (
        // pauseConfigSetZero sets the config to 0.
        pauseConfigSetZero pauseConfigExpectation = iota
        // pauseConfigSetMulStoresCount multiplies the existing value by
        // number of stores. The value is limited to 40, as larger value
        // may make the cluster unstable.
        pauseConfigMulStoresCount
        // pauseConfigSetFalse sets the config to "false".
        pauseConfigSetFalse
    )
    ```

2. change the type `expectPDCfg` to `map[string]pauseConfigExpectation`
3. change this switch to 

    ```go
    switch cfgVal {
    case pauseConfigSetZero:
        disablePDCfg[cfgKey] = 0
    case pauseConfigSetFalse:
        // this has to be a string instead of a boolean because <...>
        disablePDCfg[cfgKey] = "false"
    case pauseConfigSetMulStoresCount:
        ...
    }
    ```